src/components/ValueSet/ValueSetForm.tsx

```diff
@@ -319,14 +337,29 @@ export function ValueSetForm({
         z.object({
           system: z.string(),
           concept: z
-            .array(z.object({ code: z.string(), display: z.string() }))
+            .array(
+              z.object({
+                code: z.string().refine((val) => val !== "", {
+                  message: t("code_required"),
+                }),
+                display: z.string().refine((val) => val !== "", {
+                  message: t("display_required"),
+                }),
+              }),
+            )
             .optional(),
           filter: z
             .array(
               z.object({
-                property: z.string(),
-                op: z.string(),
-                value: z.string(),
+                property: z.string().refine((val) => val !== "", {
+                  message: t("property_required"),
+                }),
+                op: z.string().refine((val) => val !== "", {
+                  message: t("operator_required"),
+                }),
+                value: z.string().refine((val) => val !== "", {
+                  message: t("value_required"),
```

Comment:
just re-use the existing `field_required` instead of keeping the error messages very specific for each field!


```suggestion
                  message: t("field_required"),
                }),
                op: z.string().refine((val) => val !== "", {
                  message: t("field_required"),
                }),
                value: z.string().refine((val) => val !== "", {
                  message: t("field_required"),
```